### PR TITLE
Add Pod Security Policies feature

### PR DESCRIPTION
**Pull request by davidphay**
Add Pod Security Policies feature

### CODE REVIEW

File: templates/puppetdb-podsecuritypolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    ranges:
+      - min: 1
+        max: 65535
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetdb-podsecuritypolicy.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetdb-role.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: {{ .Release.Namespace }}
+rules:
+  {{- if .Values.puppetdb.psp.create }}
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    verbs: ["use"]
+    resourceNames:
+      - {{ default "puppetdb" .Values.puppetdb.serviceAccount.accountName }}-psp
+  {{- end }}
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetdb-role.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetdb-rolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ default "puppetdb" .Values.puppetdb.serviceAccount.accountName }}-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ default "puppetdb" .Values.puppetdb.serviceAccount.accountName }}
+    namespace: {{ .Release.Namespace }}
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetdb-rolebinding.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetdb-serviceaccount.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{{- if and (.Values.puppetdb.serviceAccount.enabled) (.Values.puppetdb.serviceAccount.create) }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.puppetdb.serviceAccount.accountName | default "puppetdb" }}
+  namespace: {{ .Release.Namespace }}
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetdb-serviceaccount.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetserver-podsecuritypolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    ranges:
+      - min: 1
+        max: 65535
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetserver-podsecuritypolicy.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetserver-role.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    helm.sh/hook-weight: "-10"
+rules:
+  {{- if .Values.puppetserver.psp.create }}
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    verbs: ["use"]
+    resourceNames:
+      - {{ default "puppetserver" .Values.puppetserver.serviceAccount.accountName }}-psp
+  {{- end }}
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetserver-role.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetserver-rolebinding.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    helm.sh/hook-weight: "-10"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ default "puppetserver" .Values.puppetserver.serviceAccount.accountName }}-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ default "puppetserver" .Values.puppetserver.serviceAccount.accountName }}
+    namespace: {{ .Release.Namespace }}
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetserver-rolebinding.yaml]: Could you please add an empty line here as EOF?

File: templates/puppetserver-serviceaccount.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.puppetserver.serviceAccount.accountName | default "puppet" }}
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: before-hook-creation
+    helm.sh/hook-weight: "-10"
+{{- end }}
```
Comment (Xtigyro) [inline templates/puppetserver-serviceaccount.yaml]: Could you please add an empty line here as EOF?

**Review by Xtigyro — CHANGES_REQUESTED** [review]
Thanks for the contribution, @davidphayanouvong! 🥇

Could you please update the `CHANGELOG.md` following the same pattern for previous releases and bump the minor chart version in `Chart.yaml`?

I've also requested a few other very minor changes.

Ah, yeah - finally, you might want to include your name in the [Credits](https://github.com/puppetlabs/puppetserver-helm-chart#credits)? 🥇

**Comment by davidphayanouvong** [comment]
I fixed all recommendation and updated the changelog !

**Comment by davidphay** [comment]
@Xtigyro I fixed all recommendations and updated the changelog!

**Review by Xtigyro — APPROVED** [review]
Thank you very much once more, @davidphayanouvong - this is a very nice PR!

**Pull request merged**